sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -164,4 +166,19 @@ object SQLExecution {
       }
     }
   }
+
+  /**
+   * Wrap passed function to ensure necessary thread-local variables like
+   * SparkContext local properties are forwarded to execution thread
+   */
+  def withThreadLocalCaptured[T](sparkSession: SparkSession, exec: ExecutionContext)(
+    body: => T): Future[T] = {
+    val activeSession = sparkSession
```

Comment:
Shall we have `val sc = sparkSession.sparkContext` here?

Reply:
done